### PR TITLE
Oldstation now has their own RND

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

This PR does many things and I expect to be asked to atomize some stuff.

### Oldstation Additions

Oldstation now has their own research server, generating their own points. To help alleviate major concerns, they have a few stuff to help with this:
- They now have a pre-built operating computer
- They now have an Autopsy scanner
- They now have access to Experimental Dissection

Experimental Dissection is the old dissection, giving research points in the form of paper notes on completion. They must be turned in to the RND server (only works on the Oldstation one, so you can't abuse this) for points. This was an idea I've had for some time, as Oldstation is used somewhat as a representation of how ss13 used to be (through its use of directional windows (before they got removed, but I'd like to bring them back), old IDs, RTGs, and old engines before they got removed from the game fully)

Considering there are 11 alien mobs in Oldstation, there is 27.5k research points to get from alien corpses, enough to bring them up to speed with whatever they wish to do. This is their 'alternative' for experiments (which they can still do if they want, but it is very hard to do, outside of dissection which is needed for the node).

This surgery isn't repeatable, isn't upgradable, and isn't removed by being healed. It is not mutually exclusive with autopsy (so you can't ruin yourself doing the wrong surgery).

### Other stuff

- Ninjas now drain RND points from the server they drain from, rather than always hitting science
- Syncing machines to research roundstart is now a macro, and now immediately syncs to a server on your Z-level. Machines will automatically connect to the Science servers if none else are available.

### non-player facing

- Science, Admin, and Oldstation techwebs are now no longer vars on research, but stored in research's list of techwebs.
- ``get_available_servers`` and ``find_valid_servers`` are now procs on the research subsystem, rather than the experisci component.
- Oldstation code has been split into separate files.

## Why It's Good For The Game

Oldstation is one of my favorite ruins, but it is also one of the largest complaints for RND (along with Golems) because they use the station's Science nodes & points (I recently tried de-hardcoding Science stuff to help prepare for this, but I didn't get everything in advance).

The complaint stems from these ghost roles, who are meant to be a fun activity to do while waiting for the next round, using the station's research points for their own stuff, completely untrackable unless someone goes out of their way to grief a ghost spawn just for using points to get things they need. These roles make their own servers to drain the station, and I find that unfun and quite boring for everyone- it's also not very flavorful, why would Charliestation know of the station's RND to take advantage of it?

This hopes to fix those issues, make Charliestation more worthwhile, and more flavorful.

## Changelog

:cl:
fix: Getting a node researched now properly makes it no longer hidden.
fix: Ninjas draining RD servers now drains it from the connected techweb, rather than sniping Science.
balance: Machines will first try to connect to a techweb with servers on their z-level, with the Science techweb remaining as fallback.
add: Oldstation RND, comes with their own Techweb and special surgery to gain research points through dissecting Xenomorphs.
/:cl:

### CODE REVIEW

**Comment by iwishforducks** [comment]
Really good stuff. Golems and oldstation having to rely on the station's research was a pain in the arse for all parties. Letting oldstation do the old dissection experiment is also a lot of fun.

**Comment by JohnFulpWillard** [comment]
This doesn't give Golems their own RD server but it is an easy addition after this

**Comment by CRITAWAKETS** [comment]
Currently drones and the cybersun ship (if that's still a thing i might just have lingering mapfiles) also need techwebs, so it might be good to do something like z-levels without an "alternate" network defaulting to the station network.

**Comment by JohnFulpWillard** [comment]
I added the science techweb as a fallback. It would be nice if it was temporary but I guess it'll have to see how it goes.

File: code/game/machinery/computer/dna_console.dm [hunk 1]
```diff
@@ -235,10 +235,13 @@
 	// Set the default tgui state
 	set_default_state()
 
+/obj/machinery/computer/scan_consolenew/LateInitialize()
```
Comment (Jacquerel) [inline code/game/machinery/computer/dna_console.dm:238]: oh god what a typepath

**Pull request merged**